docs/proposals/mqtt.md

```diff
@@ -0,0 +1,234 @@
+
+      > NOTE: In a well developed MQTT Application, IoT Devices should provide a Will message which the MQTT Monitoring Broker can use to detect if these
+         devices are still available and their related applications should be stopped or kept-running
+
+2. After determining if the MQTT CONNECT message is not malformed, the MQTT Broker Server sends an Acknowledgement back the Monitoring Broker with 
+   MQTT CONNACK command packet.
+
+3. Upon successful connection with the MQTT Broker Server, the Monitoring Broker can subscribe to a list of MQTT topics with possible Quality of Service
+   (QoS) parameter either set to `1` or `2` depending on the guarantees required (generally, a QoS of 1 should work)
+
+4. The Broker Server then sends back an Acknowledgement of the requested subscriptions via the SUBACK packet. This packet contains return codes for all the
+   list of topics and their QoS. 
+
+5. If the IoT Devices already have been publishing data prior to connection of Monitoring Broker with an assumption that they publish their data with QoS 1,
+   the Broker Server would publish all the information to the Monitoring Broker (persistent storage). If IoT Devices are yet to publish on their dedicated
+   topics, the MQTT Broker Server will publish data to the Monitoring Broker once the data is available.
+
+A possible integration of Akri's MQTT Monitoring Broker and MQTT Broker Server looks as follows:
+
+![Akri MQTT Monitoring Broker and MQTT Broker Server System Diagram](../media/akri_mqtt_broker_server.png)
+
+
+## Questions
+
+- Would a generic MQTT Monitoring Broker be possible to implement with Akri?
+    - If not, maybe would it be wise to begin with a Specification like [Sparkplug MQTT Specification](https://sparkplug.eclipse.org) where the Topic Structure
+      and the Payload are defined?
+
+- Is gRPC Server required within the MQTT Monitoring Broker? Does the Publish/Subscribe mechanism require a different technology?
+
+- How actually would Discovery Handler Look like? When one talks about Discovery Handler, is this pertaining to only MQTT Broker Server or the discovery of leaf
+  devices in this scenario?
+
+- Does the Integration of Akri with MQTT makes sense with already deployed MQTT Broker Servers on the Cloud or On-Premise Private Clouds or does it pertains only
+  to Edge Devices where such MQTT Broker Servers are deployable e.g. Single-Board Computers (Raspberry Pi 4, Jetson Nano etc.),
+  Small Form-Factor Computers like (Intel NUCs etc.)
+
+## Miscellany
+
+### Filters
+
+- The only possible filters that might be available for User are inclusion of a MQTT Topics List.
+
+For example a possible draft MQTT configuration to be applied to the cluster as:
+
+```yaml
+apiVersion: akri.sh/v0
+kind: Configuration
+metadata:
+  name: mqtt
+spec:
+  discoveryDetails:
+    name: mqtt
+    discoveryDetails: |+
+      serverUrls:
```

Comment:
What are the server urls? The urls to the MQTT brokers? Is there a way to discover them without being given their addresses? If you are modeling after OPC UA, the server urls can be explicitly provided but ideally that implementation will be expanded to do a network scan instead. Is something like that possible for MQTT?

Reply:
There is no implementation MQTT Broker discovery. The only possible way to achieve this is AFAIK is through [Zeroconf/mDNS/DNS-SD Solution](https://github.com/mqtt/mqtt.org/wiki/broker_auto-discovery). Possible scan on standard ports like 1883 or 8883 might yield results of possible MQTT Brokers on devices however this is not standard practice.

There is however an alternative Protocol called [MQTT for Sensor Networks aka. MQTT-SN](https://www.oasis-open.org/committees/download.php/66091/MQTT-SN_spec_v1.2.pdf) which is an OASIS Standard which tackles the issue of making MQTT-SN more compatible for Low Power Devices in Lossy Networks that do not necessarily use TCP/IP Stack e.g. ZigBee, Bluetooth, LoRa etc.